packages/core/src/lib/sharding.ts

```diff
@@ -0,0 +1,10 @@
+import type { PeerId } from "@libp2p/interface/peer-id";
+import type { PeerStore } from "@libp2p/interface/peer-store";
+
+export async function getPeerShardInfo(
+  peerId: PeerId,
+  peerStore: PeerStore
+): Promise<Uint8Array | undefined> {
```

Comment:
as discussed this can leave in `connection_manager.ts` and should return `ShardInfo`.

At a later stage, it may make sense to create a `PeerStore` that is set at `Waku.peerStore` and provided added functionality such as getting and setting shardinfo.